### ISSUE TITLE
Segfault calling layerWithIdentifier after initializing MGLMapView

### ISSUE THREAD

**Issue by waphumanoid** [mapbox/mapbox-gl-native#6361]
Current master branch, iOS platform.
If I call layerWithIdentifier method right after _mapView init, it will crash the app.
```
NSURL *styleURL = [NSURL URLWithString:@"http://...json"];// = [NSURL URLWithString:@"demo_styles.json"];
_mapView = [[MGLMapView alloc] initWithFrame:self.view.bounds styleURL:styleURL];
[self.view addSubview:_mapView];
MGLLineStyleLayer *roadLayer = (MGLLineStyleLayer *)[_mapView.style layerWithIdentifier:@"stagnation"];
```

Crash EXC_BAD_ACCESS in
```
template <>
inline bool Layer::is<FillLayer>() const {
    return type == Type::Fill;
}
```
called from 
```
- (id <MGLStyleLayer>)layerWithIdentifier:(NSString *)identifier
{
    auto layer = self.mapView.mbglMap->getLayer(identifier.UTF8String);
    
    Class clazz = [self classFromLayer:layer]; <---  layer = NULL
```

**Comment by 1ec5** [comment]
It isn’t possible to reliably interact with a style immediately after initializing the map, since style loading is asynchronous. The most reliable place to get the road layer would be in `-[MGLMapViewDelegate mapViewDidFinishLoadingMap:]`. We should throw an exception or fail gracefully rather than segfaulting like this, though.

Related: #6121.

**Comment by 1ec5** [comment]
This will be fixed by #6408.
